src/lib/mina_transition/external_transition.ml

```diff
@@ -858,7 +858,8 @@ let skip_protocol_versions_validation
   (t, Validation.Unsafe.set_valid_protocol_versions validation)
 
 module With_validation = struct
-  let compare (t1, _) (t2, _) = compare (With_hash.data t1) (With_hash.data t2)
+  let compare (t1, _) (t2, _) =
+    Poly.compare (With_hash.data t1) (With_hash.data t2)
```

Comment:
This seems like a case where we shouldn't have been using polymorphic comparison in the first place. Maybe we should just fix this spot up.